source/Body.cpp

```diff
@@ -55,22 +62,50 @@ Body::Body(const Body &sprite, Point position, Point velocity, Angle facing, dou
 // Check that this Body has a sprite and that the sprite has at least one frame.
 bool Body::HasSprite() const
 {
+	const Sprite *sprite = GetSprite();
+	return (sprite && sprite->Frames());
+}
+
+
+
+// Check that this Body has a sprite for a specific BodyState.
+bool Body::HasSpriteFor(Body::BodyState state) const
+{
+	const Sprite *sprite = sprites[state].GetSprite(0);
 	return (sprite && sprite->Frames());
 }
 
 
 
 // Access the underlying Sprite object.
-const Sprite *Body::GetSprite() const
+const Sprite *Body::GetSprite(Body::BodyState state) const
 {
-	return sprite;
+	Body::BodyState selected = state != Body::BodyState::CURRENT ? state : currentState;
+
+	SpriteParameters *spriteState = &sprites[selected];
+
+	// Return flying sprite if the requested state's sprite does not exist.
+	const Sprite *sprite = spriteState->GetSprite();
+	if(sprite != nullptr && !returnDefaultSprite)
+		return sprite;
+	else
+		return sprites[Body::BodyState::FLYING].GetSprite();
+
+}
+
+
+
+Body::BodyState Body::GetState() const
```

Comment:
```suggestion
BodyState Body::GetState() const
```

Reply:
Hitting a compilation error when making this change for some reason.

Reply:
ah yes sorry, forgot that here you are technically outside the class